packages/replay/src/auth.ts

```diff
@@ -263,3 +282,115 @@ export async function maybeAuthenticateUser(options: Options = {}) {
     return false;
   }
 }
+
+async function getAuthInfo(key: string): Promise<string> {
+  const resp = await query(
+    "AuthInfo",
+    `
+        query AuthInfo {
+          viewer {
+            user {
+              id
+            }
+          }
+          auth {
+            workspaces {
+              edges {
+                node {
+                  id
+                }
+              }
+            }
+          }
+        }
+      `,
+    undefined,
+    key
+  );
+
+  if (resp.errors) {
+    throw new GraphQLError("Failed to fetch auth info", resp.errors);
+  }
+
+  const response = resp.data as {
+    viewer: {
+      user: {
+        id: string | null;
+      } | null;
+    };
+    auth: {
+      workspaces: {
+        edges: {
+          node: {
+            id: string;
+          };
+        }[];
+      };
+    };
+  };
+
+  const { viewer, auth } = response;
+
+  if (viewer?.user?.id) {
+    viewer.user.id;
+  }
+
+  if (auth?.workspaces?.edges?.[0]?.node?.id) {
+    return auth.workspaces.edges[0].node.id;
```

Comment:
We lost here the information about the type of this id. I assume it's not needed right now but I wonder if it won't become important for some other things in the future. Since we are already fetching it - would it make sense to return the type alongside the id?

Reply:
Encoded ID has the prefix that defines the type. If we ever need it we can parse that.